src/audio/clap/plugin_manager.h

```diff
@@ -0,0 +1,79 @@
+/*
+ *  SPDX-License-Identifier: GPL-2.0-or-later
+ *
+ *  Copyright (C) 2024-2024  The DOSBox Staging Team
+ *
+ *  This program is free software; you can redistribute it and/or modify
+ *  it under the terms of the GNU General Public License as published by
+ *  the Free Software Foundation; either version 2 of the License, or
+ *  (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License along
+ *  with this program; if not, write to the Free Software Foundation, Inc.,
+ *  51 Franklin Street, Fifth Floor, Boston, MA 02110-1301, USA.
+ */
+
+#ifndef DOSBOX_CLAP_PLUGIN_MANAGER_H
+#define DOSBOX_CLAP_PLUGIN_MANAGER_H
+
+#include <string>
+#include <vector>
+
+#include "dynlib.h"
+#include "plugin.h"
+#include "std_filesystem.h"
+
+namespace Clap {
+
+struct PluginInfo {
+	std_fs::path path = {};
+
+	std::string id          = {};
+	std::string name        = {};
+	std::string description = {};
+};
```

Comment:
I think this would be a good place to distinguish between libraries and plugins.  Something like this:

```
struct Library {
   void* handle; // from dlopen()
   std::vector<PluginInfo>;
}
```

And remove the path from `PluginInfo` as that relates to the library and isn't needed if we can keep a handle instead to avoid redundant `dlopen()` calls.

Reply:
Yeah I'll do all that. Not because dlopen, I don't care about that, but because of your next comment 😄